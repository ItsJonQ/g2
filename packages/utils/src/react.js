import { Children, isValidElement } from 'react';

/**
 * Merges React `ref` together.
 */
export { default as mergeRefs } from 'react-merge-refs';

/**
 * Copies non-react specific statics from a child component to a parent component.
 * Similar to Object.assign, but with React static keywords prevented from being overridden.
 */
export { default as hoistNonReactStatics } from 'hoist-non-react-statics';

/**
 * Gets a collection of available children elements from a React component's children prop.
 *
 * @param {import('react').ReactNode} children
 *
 * @return {import('react').ReactNodeArray} An array of available children.
 */
export function getValidChildren(children) {
	if (typeof children === 'string') return [children];

	return Children.toArray(children).filter((child) => isValidElement(child));
}

/**
 * Gets the displayName of a React Component or element.
 *
 * @param {string | import('react').ComponentType} tagName
 *
 * @return {string} The display name of the Component / tagName.
 */
export function getDisplayName(tagName) {
	const displayName =
		typeof tagName === 'string'
			? tagName
			: tagName.displayName || tagName.name || 'Component';

	return displayName;
}

/**
 * Checks to see if a `children` prop is a render function prop.
 *
 * @param {*} children
 *
 * @return {boolean} True, if children is a render function prop.
 */
export function isRenderProp(children) {
	return typeof children === 'function';
}

/**
 * Handles the rendering of a React component's children prop, which
 * may be a render function prop.
 *
 * @param {*} children Children to render.
 * @param {*} props Props to pass into a (potential) children render function.
 *
 * @return {*} The rendered children.
 */
export function renderChildren(children, props = {}) {
	if (isRenderProp(children)) {
		// eslint-disable-next-line no-unused-vars
		const { children: _, ...rest } = props;
		return children(rest);
	}

	return children;
}
