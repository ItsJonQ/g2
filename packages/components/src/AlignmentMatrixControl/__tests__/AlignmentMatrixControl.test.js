import { render } from '@testing-library/react';
import React from 'react';

import { AlignmentMatrixControl } from '../index';

describe('props', () => {
	test('should render correctly', () => {
		const { container } = render(
			<AlignmentMatrixControl value="center center" />,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
