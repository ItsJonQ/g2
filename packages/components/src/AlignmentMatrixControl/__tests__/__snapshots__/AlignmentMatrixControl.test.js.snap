// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props should render correctly 1`] = `
.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0 {
  box-sizing: border-box;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  font-family: Inter,system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",sans-serif;
  font-family: var(--wp-g2-font-family);
  font-size: 13px;
  font-size: var(--wp-g2-font-size);
  font-weight: normal;
  font-weight: var(--wp-g2-font-weight);
  margin: 0;
  border: 1px solid transparent;
  border-radius: 2px;
  border-radius: var(--wp-g2-control-border-radius);
  cursor: pointer;
  display: -ms-grid;
  display: grid;
  -ms-grid-columns: auto;
  grid-template-columns: auto;
  -ms-grid-rows: (calc(92px / 3))[3];
  grid-template-rows: repeat(3,calc(92px / 3));
  outline: none;
  width: 92px;
}

@media (prefers-reduced-motion) {
  .emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0 {
    -webkit-transition: none!important;
    transition: none!important;
  }
}

[data-system-ui-reduced-motion-mode="true"] .emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0 {
  -webkit-transition: none!important;
  transition: none!important;
}

.emotion-1.emotion-1.emotion-1.emotion-1.emotion-1.emotion-1.emotion-1 {
  box-sizing: border-box;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  font-family: Inter,system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",sans-serif;
  font-family: var(--wp-g2-font-family);
  font-size: 13px;
  font-size: var(--wp-g2-font-size);
  font-weight: normal;
  font-weight: var(--wp-g2-font-weight);
  margin: 0;
  display: -ms-grid;
  display: grid;
  -ms-grid-columns: (1fr)[3];
  grid-template-columns: repeat(3,1fr);
}

@media (prefers-reduced-motion) {
  .emotion-1.emotion-1.emotion-1.emotion-1.emotion-1.emotion-1.emotion-1 {
    -webkit-transition: none!important;
    transition: none!important;
  }
}

[data-system-ui-reduced-motion-mode="true"] .emotion-1.emotion-1.emotion-1.emotion-1.emotion-1.emotion-1.emotion-1 {
  -webkit-transition: none!important;
  transition: none!important;
}

.emotion-2.emotion-2.emotion-2.emotion-2.emotion-2.emotion-2.emotion-2 {
  box-sizing: border-box;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  font-family: Inter,system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",sans-serif;
  font-family: var(--wp-g2-font-family);
  font-size: 13px;
  font-size: var(--wp-g2-font-size);
  font-weight: normal;
  font-weight: var(--wp-g2-font-weight);
  margin: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  outline: none;
  padding: 0;
  position: relative;
}

@media (prefers-reduced-motion) {
  .emotion-2.emotion-2.emotion-2.emotion-2.emotion-2.emotion-2.emotion-2 {
    -webkit-transition: none!important;
    transition: none!important;
  }
}

[data-system-ui-reduced-motion-mode="true"] .emotion-2.emotion-2.emotion-2.emotion-2.emotion-2.emotion-2.emotion-2 {
  -webkit-transition: none!important;
  transition: none!important;
}

.emotion-3.emotion-3.emotion-3.emotion-3.emotion-3.emotion-3.emotion-3 {
  box-sizing: border-box;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  font-family: Inter,system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",sans-serif;
  font-family: var(--wp-g2-font-family);
  font-size: 13px;
  font-size: var(--wp-g2-font-size);
  font-weight: normal;
  font-weight: var(--wp-g2-font-weight);
  margin: 0;
  background: currentColor;
  color: #ccd0d4;
  color: var(--wp-g2-light-gray-700);
  display: -ms-grid;
  display: grid;
  height: 6px;
  margin: auto;
  -webkit-transition: all 120ms linear;
  -webkit-transition: all var(--wp-g2-transition-duration-faster) linear;
  transition: all 120ms linear;
  transition: all var(--wp-g2-transition-duration-faster) linear;
  width: 6px;
}

@media (prefers-reduced-motion) {
  .emotion-3.emotion-3.emotion-3.emotion-3.emotion-3.emotion-3.emotion-3 {
    -webkit-transition: none!important;
    transition: none!important;
  }
}

[data-system-ui-reduced-motion-mode="true"] .emotion-3.emotion-3.emotion-3.emotion-3.emotion-3.emotion-3.emotion-3 {
  -webkit-transition: none!important;
  transition: none!important;
}

*:hover>.emotion-3.emotion-3.emotion-3.emotion-3.emotion-3.emotion-3.emotion-3 {
  color: #007cba;
  color: var(--wp-g2-color-admin);
}

.emotion-12.emotion-12.emotion-12.emotion-12.emotion-12.emotion-12.emotion-12 {
  box-sizing: border-box;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  font-family: Inter,system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",sans-serif;
  font-family: var(--wp-g2-font-family);
  font-size: 13px;
  font-size: var(--wp-g2-font-size);
  font-weight: normal;
  font-weight: var(--wp-g2-font-weight);
  margin: 0;
  background: currentColor;
  color: #ccd0d4;
  color: var(--wp-g2-light-gray-700);
  display: -ms-grid;
  display: grid;
  height: 6px;
  margin: auto;
  -webkit-transition: all 120ms linear;
  -webkit-transition: all var(--wp-g2-transition-duration-faster) linear;
  transition: all 120ms linear;
  transition: all var(--wp-g2-transition-duration-faster) linear;
  width: 6px;
  box-shadow: 0 0 0 2px #000000;
  box-shadow: 0 0 0 2px var(--wp-g2-black);
  color: #000000;
  color: var(--wp-g2-black);
}

@media (prefers-reduced-motion) {
  .emotion-12.emotion-12.emotion-12.emotion-12.emotion-12.emotion-12.emotion-12 {
    -webkit-transition: none!important;
    transition: none!important;
  }
}

[data-system-ui-reduced-motion-mode="true"] .emotion-12.emotion-12.emotion-12.emotion-12.emotion-12.emotion-12.emotion-12 {
  -webkit-transition: none!important;
  transition: none!important;
}

*:hover>.emotion-12.emotion-12.emotion-12.emotion-12.emotion-12.emotion-12.emotion-12 {
  color: #007cba;
  color: var(--wp-g2-color-admin);
}

*:hover>.emotion-12.emotion-12.emotion-12.emotion-12.emotion-12.emotion-12.emotion-12 {
  color: #000000;
  color: var(--wp-g2-black);
}

<div
  aria-label="Alignment Matrix Control"
  class="components-alignment-matrix-control wp-components-alignment-matrix-control ic-o4vf4i ic-otytyz ic-192yz5d emotion-0"
  data-g2-c16t="true"
  data-g2-component="AlignmentMatrixControl"
  id="alignment-matrix-control"
  role="grid"
>
  <div
    class="ic-1blhbvo ic-192yz5d emotion-1"
    id="alignment-matrix-control-1"
    role="row"
  >
    <span
      class="ic-151osxh ic-192yz5d emotion-2"
      id="alignment-matrix-control-top-left"
      role="gridcell"
      tabindex="-1"
    >
      <span
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
      >
        top left
      </span>
      <span
        aria-describedby="alignment-matrix-control-top-left"
        class="ic-1c3kh7b ic-192yz5d emotion-3"
        role="presentation"
      />
    </span>
    <span
      class="ic-151osxh ic-192yz5d emotion-2"
      id="alignment-matrix-control-top-center"
      role="gridcell"
      tabindex="-1"
    >
      <span
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
      >
        top center
      </span>
      <span
        aria-describedby="alignment-matrix-control-top-center"
        class="ic-1c3kh7b ic-192yz5d emotion-3"
        role="presentation"
      />
    </span>
    <span
      class="ic-151osxh ic-192yz5d emotion-2"
      id="alignment-matrix-control-top-right"
      role="gridcell"
      tabindex="-1"
    >
      <span
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
      >
        top right
      </span>
      <span
        aria-describedby="alignment-matrix-control-top-right"
        class="ic-1c3kh7b ic-192yz5d emotion-3"
        role="presentation"
      />
    </span>
  </div>
  <div
    class="ic-1blhbvo ic-192yz5d emotion-1"
    id="alignment-matrix-control-5"
    role="row"
  >
    <span
      class="ic-151osxh ic-192yz5d emotion-2"
      id="alignment-matrix-control-center-left"
      role="gridcell"
      tabindex="-1"
    >
      <span
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
      >
        center left
      </span>
      <span
        aria-describedby="alignment-matrix-control-center-left"
        class="ic-1c3kh7b ic-192yz5d emotion-3"
        role="presentation"
      />
    </span>
    <span
      class="ic-151osxh ic-192yz5d emotion-2"
      id="alignment-matrix-control-center-center"
      role="gridcell"
      tabindex="0"
    >
      <span
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
      >
        center center
      </span>
      <span
        aria-describedby="alignment-matrix-control-center-center"
        class="ic-1c3kh7b ic-192yz5d emotion-12"
        role="presentation"
      />
    </span>
    <span
      class="ic-151osxh ic-192yz5d emotion-2"
      id="alignment-matrix-control-center-right"
      role="gridcell"
      tabindex="-1"
    >
      <span
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
      >
        center right
      </span>
      <span
        aria-describedby="alignment-matrix-control-center-right"
        class="ic-1c3kh7b ic-192yz5d emotion-3"
        role="presentation"
      />
    </span>
  </div>
  <div
    class="ic-1blhbvo ic-192yz5d emotion-1"
    id="alignment-matrix-control-9"
    role="row"
  >
    <span
      class="ic-151osxh ic-192yz5d emotion-2"
      id="alignment-matrix-control-bottom-left"
      role="gridcell"
      tabindex="-1"
    >
      <span
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
      >
        bottom left
      </span>
      <span
        aria-describedby="alignment-matrix-control-bottom-left"
        class="ic-1c3kh7b ic-192yz5d emotion-3"
        role="presentation"
      />
    </span>
    <span
      class="ic-151osxh ic-192yz5d emotion-2"
      id="alignment-matrix-control-bottom-center"
      role="gridcell"
      tabindex="-1"
    >
      <span
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
      >
        bottom center
      </span>
      <span
        aria-describedby="alignment-matrix-control-bottom-center"
        class="ic-1c3kh7b ic-192yz5d emotion-3"
        role="presentation"
      />
    </span>
    <span
      class="ic-151osxh ic-192yz5d emotion-2"
      id="alignment-matrix-control-bottom-right"
      role="gridcell"
      tabindex="-1"
    >
      <span
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
      >
        bottom right
      </span>
      <span
        aria-describedby="alignment-matrix-control-bottom-right"
        class="ic-1c3kh7b ic-192yz5d emotion-3"
        role="presentation"
      />
    </span>
  </div>
</div>
`;
